types/vote.go

```diff
@@ -353,6 +374,15 @@ func (vote *Vote) ValidateBasic() error {
 		if len(vote.ExtensionSignature) == 0 && len(vote.Extension) != 0 {
 			return ErrVoteNoSignature
 		}
+		if len(vote.NonRpExtensionSignature) == 0 && len(vote.NonRpExtension) != 0 {
+			return errors.New("vote extension signature absent on vote with extension")
+		}
+
+		// Vote extensions and non replay protected vote extensions must go together
```

Comment:
I'm curious why that is.

Reply:
"These non replay-protected vote extensions are optional."

Reply:
They go together in the sense that
* if vote extensions are disabled in `ConsensusParams`, both vote extensions are empty and signatures are empty
* if vote extensions are enabled in `ConsensusParam`, both signatures are present, even if either of the vote extensions field is empty. If a field is empty, the signature should verify against the empty byte slice.

@bermuell just clarified that in the doc (check the latest commit)